### PR TITLE
Single DML POST API for all operations with Multi Collection Support

### PR DESCRIPTION
**Pull request by tushardhadiwal**

### CODE REVIEW

**Comment by jessebenson** [comment]
@jdhuntington @andrewjamconrad

File: src/Microsoft.ServiceFabric.Services.Queryable/Controller/QueryableController.cs [hunk 1]
```diff
@@ -170,121 +111,72 @@ private void AddAccessControlHeaders(HttpRequestMessage request, HttpResponseMes
...
+							tempResult.PartitionId = mypid;
+
+							finalResult.Add(tempResult);
+						}
+						listOfStatusCodes = await proxy.DmlAsync(backendObjects.ToArray());
+						foreach (var row in listOfStatusCodes)
+						{
+							if (p < finalResult.Count)
+							{
+								finalResult[p].Status = row; ;
```
Comment (jessebenson) [inline src/Microsoft.ServiceFabric.Services.Queryable/Controller/QueryableController.cs]: (nit): extra semi-colon.

File: src/Microsoft.ServiceFabric.Services.Queryable/Query/DmlResult.cs [hunk 2]
```diff
@@ -1,13 +1,14 @@
...
-using System.Collections.Generic;
-
+using System;
 
 namespace Microsoft.ServiceFabric.Services.Queryable
 {
 	public class DmlResult
 	{
+		[JsonProperty("Collection")]
+		public string collection { get; set; }
```
Comment (jessebenson) [inline src/Microsoft.ServiceFabric.Services.Queryable/Query/DmlResult.cs]: (nit):  can you make this capitalized "Collection"

File: src/Microsoft.ServiceFabric.Services.Queryable/Controller/QueryableController.cs [hunk 3]
```diff
@@ -170,121 +111,72 @@ private void AddAccessControlHeaders(HttpRequestMessage request, HttpResponseMes
...
-				{
-					List<int> templist = new List<int>();
-					if (preMap.ContainsKey(obj[i].PartitionId))
-					{
-						templist = preMap[obj[i].PartitionId];
-						templist.Add(i);
-						preMap[obj[i].PartitionId] = templist;
-					}
-					else
+					for (int i = 0; i < obj.Length; i++)
```
Comment (jessebenson) [inline src/Microsoft.ServiceFabric.Services.Queryable/Controller/QueryableController.cs]: You can replace the majority of this code (e.g. remove this for loop and 'preMap') with:

var partitionMap = obj.GroupBy(o => o.PartitionId);
foreach (var partition in partitionMap)
{
    var partitionId = partition.Key;
    foreach (ValueViewModel item in partition)
    {
         // ... code here to process one item within 'partitionId'
    }
}

File: src/Microsoft.ServiceFabric.Services.Queryable/Controller/ValueViewModel.cs [hunk 4]
```diff
@@ -5,6 +5,8 @@ namespace Microsoft.ServiceFabric.Services.Queryable.Controller
 {
 	public class ValueViewModel
 	{
+		public string Operation { get; set; }
```
Comment (jessebenson) [inline src/Microsoft.ServiceFabric.Services.Queryable/Controller/ValueViewModel.cs]: You should make Operation an enum.  The front-end just has to send strings just like it does.  Json (Newtonsoft) will deserialize it correctly.

public enum Operation
{
     Add,
     Update,
     Delete,
}

**Review by jessebenson — CHANGES_REQUESTED** [review]

File: src/Microsoft.ServiceFabric.Services.Queryable/Controller/QueryableController.cs (hunk 3 above)
Comment (tushardhadiwal) [reply src/Microsoft.ServiceFabric.Services.Queryable/Controller/QueryableController.cs]: @jessebenson  Sometimes partitionID does not exist in incoming data, in that case I generate a random partitionId and assign it right there, and then grouping happens. And for that we anyways have to loop at least  once through the data, so we might as well map it in the same loop to keep it as efficient as possible. What do you think?

File: src/Microsoft.ServiceFabric.Services.Queryable/Query/DmlResult.cs (hunk 2 above)
Comment (tushardhadiwal) [reply src/Microsoft.ServiceFabric.Services.Queryable/Query/DmlResult.cs]: Done.

File: src/Microsoft.ServiceFabric.Services.Queryable/Controller/QueryableController.cs (hunk 1 above)
Comment (tushardhadiwal) [reply src/Microsoft.ServiceFabric.Services.Queryable/Controller/QueryableController.cs]: Done

File: src/Microsoft.ServiceFabric.Services.Queryable/Controller/ValueViewModel.cs (hunk 4 above)
Comment (tushardhadiwal) [reply src/Microsoft.ServiceFabric.Services.Queryable/Controller/ValueViewModel.cs]: Done.

**Pull request merged**